### PR TITLE
Add compatibility with Yunohost 3.2 / Stretch:

### PR DESCRIPTION
**Pull request by klorydryk**
* Pip<10
* Nginx path configuration dynamic

### CODE REVIEW

**Review by pitchum — APPROVED** [review]

File: conf/nginx.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 location PATHTOCHANGE/static/ {
-    alias   /opt/yunohost/ihatemoney/venv/lib/python3.4/site-packages/ihatemoney/static/;
+    alias   APPHOME/static/;
```
Comment (JocelynDelalande) [inline conf/nginx.conf:2]: Is that backward compatible ? If not, I think there is something to set in manifest to tell that it is only compatible ynh >=3.

File: scripts/install [hunk 2]
```diff
@@ -62,11 +62,13 @@ create_unix_user
 
 # Prepare venv
 init_virtualenv
+sudo /opt/yunohost/ihatemoney/venv/bin/pip install 'pip<10'
```
Comment (JocelynDelalande) [inline scripts/install:65]: Why pip<10 ?

(nitpicking : push it to the next line ;-)

File: scripts/_common.sh [hunk 3]
```diff
@@ -41,8 +41,10 @@ init_virtualenv () {
 configure_nginx () {
     local domain=$1
     local path=$2
+    local app_home=$3
```
Comment (JocelynDelalande) [inline scripts/_common.sh:44]: How this new argument will behave with upgrade script, where `configure_nginx` is also called ?

**Review by JocelynDelalande — COMMENTED** [review]
I am really happy to see that you managed to spare some time to make this happening :-)

I made a first review « with the eyes » to start discussion, I would like to test it for real later :-).

File: scripts/install (hunk 2 above)
Comment (klorydryk) [reply scripts/install:65]: When trying to install ihatemoney package with pip, I get the error "ImportError: No module named 'pip.req'". It appears this error comes with pip10's changes in the api.

Comment (JocelynDelalande) [reply scripts/install:65]: Oh yeah, I remember now…That is linked to https://github.com/spiral-project/ihatemoney/issues/340

In recent versions of Ihatemoney, this is fixed, but we are still on an old version.

IHM update itself is for another PR another time.

Well spotted :-)

**Comment by JocelynDelalande** [comment]
@klorydryk just curious, did you ran on any infinite redirection issue installing on https domain ? I did with latest release of ihatemoney on ynh 2.x. Like @Glandos [did](https://github.com/spiral-project/ihatemoney/pull/358).

(sadly, the fix is not released yet, but [working on it](https://github.com/spiral-project/ihatemoney/issues/396))

**Comment by klorydryk** [comment]
> @klorydryk just curious, did you ran on any infinite redirection issue installing on https domain ? I did with latest release of ihatemoney on ynh 2.x. Like @Glandos [did](https://github.com/spiral-project/ihatemoney/pull/358).
> 
> (sadly, the fix is not released yet, but [working on it](https://github.com/spiral-project/ihatemoney/issues/396))

Nope, no problem like that. Installed on my Yunohost 3.2 since, with https domain (let's encrypt).

File: conf/nginx.conf (hunk 1 above)
Comment (klorydryk) [reply conf/nginx.conf:2]: I think not, I haven't worked on the update :s

Comment (JocelynDelalande) [reply conf/nginx.conf:2]: @klorydryk OK :) do you plan to in this PR ?

**Comment by JocelynDelalande** [comment]
> Nope, no problem like that. Installed on my Yunohost 3.2 since, with https domain (let's encrypt).

Lucky you. I will check issue reproducability, but maybe it affects only 2.x

**Comment by alexAubin** [comment]
(Bumpity bump :s)

**Comment by JocelynDelalande** [comment]
v3.0 was released recently, before that, ihatemoney had a bug preventing it to work with ynh 3.x. So, advancing slowly :-).

**Comment by JocelynDelalande** [comment]
@klorydryk I found time and motivation, so I made it work against ynh 3.x (upgrade, backup… included). 

I took some inspiration from your work here, thanks :-).

Even if I tested it myself, tests of master branch on ynh 3.x still appreciated :-).

**Pull request closed without merging**